File: C/22_voltmeter_1/main.c
/*
* File name   : main.c
* Description : Read voltage from ADC0832, and display on segment display
* Website     : www.adeept.com
* E-mail      : support@adeept.com
* Author      : Jason
* Date        : 2016/06/18
*/
#include "segment.h"
#include "ADC0832.h"
#include <stdio.h>

int main(void)
{
	uchar tmp;
	float vol;

	if(wiringPiSetup() < 0){
		printf("wiringPi setup failed!\n");
		return -1;
	}

	TM1638_init();
	ADC0832_Init();
	numberDisplay(1234);
	delay(2000);

	while(1){
		tmp = get_ADC_Result();
		vol = 5.0/255 * tmp;
		printf("voltage : %.2fV\n", vol);
		numberDisplay_dec(vol);
		delay(100);
	}

	return 0;
}

File: Python/22_voltmeter_1/main.py
#!/usr/bin/env python

import RPi.GPIO as GPIO
import ADC0832
import segment
import time

def destory():
	GPIO.cleanup()

def setup():
	segment.TM1638_init()
	ADC0832.setup()

def loop():
	while True:
		t = ADC0832.getResult()
		vol = 5.0/255 * t
		vol = round(vol, 2)
		segment.numberDisplay_dec(vol)
		time.sleep(0.1)      

if __name__ == '__main__':
	print 'www.adeept.com'
	setup()
	try:
		loop()
	except KeyboardInterrupt:
		destory()
